### PR TITLE
chore: uploads mongo-runner logs and address addArb flakes MONGOSH-1573

### PR DESCRIPTION
**Pull request by mcasimir**
- Adds an `id` argument to `startTestServer` and `startTestCluster` that is used to build  the `tmpDir` and `logDir` paths for `mongodb-runner`.
- Uploads the `logDir`s as an tarball on `post:`

The `tmp/` folder is now structured as follows:

```
tmp/mongodb-runner/logs/shell-api-getPrintableShardStatus-0/... 
tmp/mongodb-runner/dbs/shell-api-getPrintableShardStatus-0/...
```

Screenshot 2023-08-24 at 09 24 16

EVG patch: https://spruce.mongodb.com/version/64e70c5a32f417b896314b27/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

### CODE REVIEW

**Review by lerouxb — APPROVED** [review]

File: testing/integration-testing-hooks.ts [hunk 1]
```diff
@@ -192,9 +212,9 @@ export function startTestServer(shareMode: 'shared' | 'not-shared', args: Partia
     if (Object.keys(args).length > 0) {
       throw new Error('Cannot specify arguments for shared mongod');
     }
-    server = sharedSetup ?? (sharedSetup = new MongoRunnerSetup());
+    server = sharedSetup ?? (sharedSetup = new MongoRunnerSetup(id));
```
Comment (addaleax) [inline testing/integration-testing-hooks.ts:215]: Including an id in this case doesn’t make much sense, does it? The point of the `shared` server is that it’s shared for all tests running in the same process and we don’t need to start up/tear down server instances each time, but that also means that assigning an `id` makes somewhat limited sense.

Comment (mcasimir) [reply testing/integration-testing-hooks.ts:215]: Ah that's true, yeah! Makes sense, i'll skip and use 'shared' it in that case.

Comment (mcasimir) [reply testing/integration-testing-hooks.ts]: @addaleax too much? https://github.com/mongodb-js/mongosh/pull/1611/commits/0b7a34493cfb610d2b6ed4d600403cf6715f95cb

Comment (mcasimir) [reply testing/integration-testing-hooks.ts]: Ah i have to update the comments

**Review by addaleax — APPROVED** [review]
:rocket:

**Comment by mcasimir** [comment]
@lerouxb @addaleax i've added a new commit: https://github.com/mongodb-js/mongosh/pull/1611/commits/b2f24a9eb140f9b63461f09e72247baabd786737 with a fix for MONGOSH-1573, it seems that `reconfig(cfg, {force:true})` is doing a good job here. However I'm seeing other random failures in my patch: https://spruce.mongodb.com/version/64e8ae9e2a60ed934f9b1ec6/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC, maybe i need to rebase, feel free to take over / merge while i'm out if is all good.

**Comment by addaleax** [comment]
Evergreen: https://spruce.mongodb.com/version/64ec6a275623436448262dd4/tasks

**Comment by addaleax** [comment]
Evergreen: https://spruce.mongodb.com/version/64ec78a60305b91c7dbcf594/tasks

**Pull request merged**